Core/src/Surfaces/DiscSurface.cpp

```diff
@@ -211,29 +209,29 @@ Acts::Vector2 Acts::DiscSurface::localCartesianToPolar(
 }
 
 Acts::BoundToFreeMatrix Acts::DiscSurface::boundToFreeJacobian(
-    const GeometryContext& gctx, const BoundVector& boundParams) const {
-  // Transform from bound to free parameters
-  FreeVector freeParams =
-      detail::transformBoundToFreeParameters(*this, gctx, boundParams);
+    const GeometryContext& gctx, const FreeVector& parameters) const {
   // The global position
-  const Vector3 position = freeParams.segment<3>(eFreePos0);
+  const Vector3 position = parameters.segment<3>(eFreePos0);
   // The direction
-  const Vector3 direction = freeParams.segment<3>(eFreeDir0);
-  // special polar coordinates for the Disc
-  double lrad = boundParams[eBoundLoc0];
-  double lphi = boundParams[eBoundLoc1];
-  double lcphi = std::cos(lphi);
-  double lsphi = std::sin(lphi);
-  // retrieve the reference frame
-  const auto rframe = referenceFrame(gctx, position, direction);
+  const Vector3 direction = parameters.segment<3>(eFreeDir0);
+  // The measurement frame of the surface
+  RotationMatrix3 rframeT =
+      referenceFrame(gctx, position, direction).transpose();
+  // calculate the transformation to local coordinates
+  const Vector3 pos_loc = transform(gctx).inverse() * position;
+  const double lr = perp(pos_loc);
+  const double lphi = phi(pos_loc);
```

Comment:
So this is a trade-off of having to do the trig here, or doing the trig when converting to free. I guess this is not *more* calculation, since we were indeed doing the trig when converting back to free.

Reply:
right and the disc is the only case where we need the local params. this is the same in `freeToBoundJacobian`